content/client-server-api/_index.md

```diff
@@ -13,20 +13,32 @@ clients which maintain a full local persistent copy of server state.
 ## API Standards
 
 The mandatory baseline for client-server communication in Matrix is
-exchanging JSON objects over HTTP APIs. HTTPS is recommended for
-communication, although HTTP may be supported as a fallback to support
-basic HTTP clients. More efficient optional transports will in future be
-supported as optional extensions - e.g. a packed binary encoding over
-stream-cipher encrypted TCP socket for low-bandwidth/low-roundtrip
-mobile usage. For the default HTTP transport, all API calls use a
-Content-Type of `application/json`. In addition, all strings MUST be
-encoded as UTF-8.
+exchanging JSON objects over HTTP APIs. More efficient transports may be
+specified in future as optional extensions.
+
+HTTPS is recommended for communication. The use of plain HTTP is not
+recommended outside test environments.
 
 Clients are authenticated using opaque `access_token` strings (see [Client
 Authentication](#client-authentication) for details).
 
+All `POST` and `PUT` endpoints, with the exception of
+[`POST /_matrix/media/v3/upload`](#post_matrixmediav3upload), require the
+client to supply a request body containing a (potentially empty) JSON object.
+Clients are *not* required to supply a `Content-Type` header.
```

Comment:
:( Can we at least say that clients "should" supply a `Content-Type: application/json` header?

Reply:
I'm not really sure if that gains us much, unless we're planning to mandate it sometime soon? Servers still have to assume it's json even if there is no content-type header (or, for that matter, if the content-type header says it is something other than json, like text).

Reply:
To me, "Client are *not* required to supply a `Content-Type` header.` sounds like an encouragement to not bother sending a `Content-Type`, which just ... feels wrong to me.  At least if we say that they "should" send a `Content-Type`, we're encouraging people to send proper HTTP.  Maybe "Clients should supply an appropriate `Content-Type` header, but servers should be able to handle requests that do not have one, or an incorrect one."?  Though that's quite verbose. :shrug: This isn't a blocking comment; I'm just reluctant to encourage people to be sloppy.

Reply:
I think "should" ~= "MUST", so just saying "should" implies that servers are allowed to enforce it.  So I could say this:

```suggestion
Clients should supply a `Content-Type` header of `application/json` for all requests with JSON bodies,
but this is not required.
```

but it feels a bit wordy. 

I don't really see what we gain, as guardians of the matrix spec, by discouraging people from this particular bit of sloppiness.

Reply:
> I don't really see what we gain, as guardians of the matrix spec, by discouraging people from this particular bit of sloppiness.

It feels to me that, since we're building on HTTP, we should be encouraging good HTTP behaviour, but maybe I'm being overly picky.

Your suggested wording seems fine to me, but if you don't think it's worth changing, then that's OK.